packages/api/src/middleware/acl/hasFarmAccess.js

```diff
@@ -192,7 +193,25 @@ async function fromLocations(locations) {
   if (!locations || !locations.length) {
     return {};
   }
-  const location_ids = locations ? locations.map((location) => location.location_id) : undefined;
+  const location_ids = locations
+    ? locations.map((location) => location.location_id ?? location)
+    : undefined;
+  try {
```

Comment:
Here it can just return fromLocationIds(location_ids)

Reply:
This function is part of the unused code block. https://github.com/LiteFarmOrg/LiteFarm/pull/1920

I'm just going to add the new function only for now if that's cool.